keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensorTest.java

```diff
@@ -46,35 +45,35 @@ public void setup() {
         matrixA = Nd4jDoubleTensor.create(new double[]{1, 2, 3, 4}, new long[]{2, 2});
         matrixB = Nd4jDoubleTensor.create(new double[]{1, 2, 3, 4}, new long[]{2, 2});
         scalarA = Nd4jDoubleTensor.scalar(2.0);
-        vectorA = Nd4jDoubleTensor.create(new double[]{1, 2, 3}, new long[]{3, 1});
-        vectorB = Nd4jDoubleTensor.create(new double[]{1, 2, 3}, new long[]{1, 3});
+        vectorA = Nd4jDoubleTensor.create(new double[]{1, 2, 3}, new long[]{3});
+        vectorB = Nd4jDoubleTensor.create(new double[]{1, 2, 3}, new long[]{3});
         rankThreeTensor = Nd4jDoubleTensor.create(new double[]{1, 2, 3, 4, 5, 6, 7, 8}, new long[]{2, 2, 2});
     }
 
     @Before
-    public void enableDebugModeForNaNChecking() throws Exception {
+    public void enableDebugModeForNaNChecking() {
         TensorValidator.NAN_CATCHER.enable();
         TensorValidator.NAN_FIXER.enable();
     }
 
     @After
-    public void disableDebugModeForNaNChecking() throws Exception {
+    public void disableDebugModeForNaNChecking() {
         TensorValidator.NAN_CATCHER.disable();
         TensorValidator.NAN_FIXER.disable();
     }
 
     @Test
-    public void youCannotCreateARankZeroTensor() {
-        thrown.expect(IllegalArgumentException.class);
-        thrown.expectMessage("Tensors must have rank >=2 : []");
-        DoubleTensor.create(new double[]{}, new long[]{});
+    public void youCanCreateARankZeroTensor() {
+        DoubleTensor scalar = DoubleTensor.create(new double[]{2.0}, new long[]{});
+        DoubleTensor expected = DoubleTensor.scalar(2.0);
+        assertEquals(expected, scalar);
+        assertEquals(0, scalar.getShape().length);
```

Comment:
You  can use the `TensorMatchers` throughout this file: `hasShape`, `valuesMatch`, `valuesAndShapesMatch`